python/lancedb/table.py

```diff
@@ -709,7 +709,11 @@ def create_scalar_index(self, column: str, *, replace: bool = True):
         self._dataset.create_scalar_index(column, index_type="BTREE", replace=replace)
 
     def create_fts_index(
-        self, field_names: Union[str, List[str]], *, replace: bool = False
+        self,
+        field_names: Union[str, List[str]],
+        *,
+        replace: bool = False,
+        writer_heap_size: Optional[int] = 1024 * 1024 * 1024,
```

Comment:
nit: we could do something like 50% of the system's memory and make this a fraction